Cython/Compiler/Nodes.py

```diff
@@ -3207,9 +3146,6 @@ def declare_python_arg(self, env, arg):
     def analyse_expressions(self, env):
         self.local_scope.directives = env.directives
         self.analyse_default_values(env)
-        self.analyse_annotations(env)
-        if self.return_type_annotation:
-            self.return_type_annotation = self.analyse_annotation(env, self.return_type_annotation)
```

Comment:
Why is this no longer needed? Shouldn't the nodes still be analysed? It's up to the annotation node itself to decide to do nothing, not the node that contains it.

Reply:
I've added these back in (and in ExprNodes.py too) in a slightly simplified form.